data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -137,11 +139,43 @@ protected long pos(T record) {
           CloseableIterable.transform(CloseableIterable.concat(deleteRecords), Record::copy),
           deleteSchema.asStruct());
 
-      filteredRecords = Deletes.filter(filteredRecords,
-          record -> projectRow.wrap(asStructLike(record)), deleteSet);
+      Predicate<T> isInDeleteSet = record -> deleteSet.contains(projectRow.wrap(asStructLike(record)));
+      isInDeleteSets.add(isInDeleteSet);
```

Comment:
Why doesn't this return a single predicate, `isDeleted`? Both `findEqualityDeleteRows` and `applyEqDeletes` end up producing a Predicate that determines whether a row is deleted. The only difference is that `deletedRows` and `remainingRows` are negations of each other. But those methods could just as easily use `isDeleted` and negate in `shouldKeep`.